### PR TITLE
feat(linter) Add ban_types typescript eslint rule

### PR DESCRIPTION
**Pull request by camc314**
Part of #503 -> `@typescript-eslint/ban-types`

### CODE REVIEW

File: crates/oxc_linter/src/rules/typescript/ban_types.rs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        match node.kind() {
+            AstKind::TSTypeReference(typ) => {
+                let name = match &typ.type_name {
+                    oxc_ast::ast::TSTypeName::IdentifierReference(v) => &v.name,
+                    oxc_ast::ast::TSTypeName::QualifiedName(_) => return,
+                };
+
+                match name.as_str() {
+                    "String" | "Boolean" | "Number" | "Symbol" | "BigInt" => {
+                        ctx.diagnostic(BanTypesDiagnostic::Type(
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/typescript/ban_types.rs:62]: Consider using `&'a Atom` here to reduce memory allocations

Comment (camc314) [reply crates/oxc_linter/src/rules/typescript/ban_types.rs:62]: instead of `name.to_string` or the match?

Comment (DonIsaac) [reply crates/oxc_linter/src/rules/typescript/ban_types.rs:62]: The former, I misclicked the line

Comment (camc314) [reply crates/oxc_linter/src/rules/typescript/ban_types.rs:64]: :+1: thanks. mind checking now?

**Review by Boshen — APPROVED** [review]
Thank you!

**Pull request merged**